File: cmk/gui/help.py
#!/usr/bin/env python3
# Copyright (C) 2019 Checkmk GmbH - License: GNU General Public License v2
# This file is part of Checkmk (https://checkmk.com). It is subject to the terms and
# conditions defined in the file COPYING, which is part of this source code package.

import json

from cmk.gui.config import Config
from cmk.gui.http import request, response
from cmk.gui.logged_in import user
from cmk.gui.pages import PageEndpoint, PageRegistry


def register(page_registry: PageRegistry) -> None:
    page_registry.register(PageEndpoint("ajax_switch_help", ajax_switch_help))


def ajax_switch_help(config: Config) -> None:
    state = request.var("enabled", "") != ""
    user.inline_help_as_text = state
    response.set_data(json.dumps(state))

File: packages/cmk-frontend/src/js/modules/help.ts
/**
 * Copyright (C) 2024 Checkmk GmbH - License: GNU General Public License v2
 * This file is part of Checkmk (https://checkmk.com). It is subject to the terms and
 * conditions defined in the file COPYING, which is part of this source code package.
 */
import { call_ajax } from './ajax'
import { add_class, has_class, remove_class } from './utils'

//#   .-Help Toggle--------------------------------------------------------.
//#   |          _   _      _         _____                 _              |
//#   |         | | | | ___| |_ __   |_   _|__   __ _  __ _| | ___         |
//#   |         | |_| |/ _ \ | '_ \    | |/ _ \ / _` |/ _` | |/ _ \        |
//#   |         |  _  |  __/ | |_) |   | | (_) | (_| | (_| | |  __/        |
//#   |         |_| |_|\___|_| .__/    |_|\___/ \__, |\__, |_|\___|        |
//#   |                      |_|                |___/ |___/                |
//#   '--------------------------------------------------------------------'

const SHOW_HELP_CLASS = 'inline_help_as_text'

function is_help_active() {
  return has_class(document.body, SHOW_HELP_CLASS)
}

export function toggle() {
  if (is_help_active()) {
    switch_help(false)
  } else {
    switch_help(true)
  }
}

function switch_help(how: boolean) {
  if (how) add_class(document.body, SHOW_HELP_CLASS)
  else remove_class(document.body, SHOW_HELP_CLASS)

  call_ajax('ajax_switch_help.py?enabled=' + (how ? 'yes' : ''))
  toggle_help_page_menu_icon()
}

function toggle_help_page_menu_icon() {
  const iconElement = document.querySelector('#menu_entry_inline_help cmk-dynamic-icon')
  if (!iconElement) return

  const dataAttr = iconElement.getAttribute('data') || ''
  const newData = is_help_active()
    ? dataAttr.replace('"toggle-off"', '"toggle-on"')
    : dataAttr.replace('"toggle-on"', '"toggle-off"')

  iconElement.setAttribute('data', newData)
}
